src/js/src/components/ToolboxComponents/LinkGraph.vue

```diff
@@ -1,24 +1,225 @@
+  components: {
+    VueSlider
+  },
+  mixins: [StringManipulationUtils],
   data() {
     return {
+      maxValue:new Date().getTime(),
+      minValue:788914800000,
+      sliderValues:[788914800000,new Date().getTime()],
+      getDate: v => `${this.createDateFromNumber(v)}`,
+      linkNumber:10,
+      ingoing:false,
+      loading:false,
+      domain:''
     }
   },
   mounted () {
-
+    window.addEventListener('resize', function() {
+     let height = document.getElementById('linkGraphContainer').offsetHeight - document.getElementById('graphControlsContainer').offsetHeight
+     document.getElementById('svgDiagram').setAttribute('height', height + 'px')
+    })
+    const routerQuery = this.$router.history.current.query
+    if(routerQuery.domain && routerQuery.dateStart && routerQuery.dateEnd && routerQuery.facetLimit && routerQuery.ingoing) {
+      routerQuery.domain ? this.domain = routerQuery.domain : null
+      routerQuery.dateStart ? this.sliderValues[0] = parseInt(routerQuery.dateStart) : null
+      routerQuery.dateEnd ? this.sliderValues[1] = parseInt(routerQuery.dateEnd) : null
+      routerQuery.facetLimit ? this.linkNumber = parseInt(routerQuery.facetLimit) : null
+      routerQuery.ingoing 
+        ? routerQuery.ingoing === 'true' 
+          ? this.ingoing = true 
+          : this.ingoing = false 
+        : null
+      this.loading = true
+      requestService.getLinkGraph(this.domain,this.linkNumber, this.ingoing, this.sliderValues[0], this.sliderValues[1]).then(result => (this.buildSvg(result)), error => console.log('Error, no link graph created.'))
+    }
   },
   methods: {
+    loadLinkGraph(domain) {
+      this.loading = true
+      requestService.getLinkGraph(this.domain,this.linkNumber, this.ingoing, this.sliderValues[0], this.sliderValues[1]).then(result => (this.buildSvg(result)), error => console.log('Error, no link graph created.'))
+    },
+    createDateFromNumber(date) {
+      let time = new Date(date)
+      let day = time.getDay()
+      let month = time.getMonth() >= 9 ? ((+time.getMonth()) + (+1)) : '0' + ((+time.getMonth()) + (+1))
+      let year = time.getFullYear()
+      return day + '-' + month + '/' + year
+    },
+    buildSvg(result) {
+      document.getElementById('graphContainer').innerHTML = ''
+      let _this = this
+      var width =  document.getElementById('graphContainer').offsetWidth, height = document.getElementById('graphContainer').offsetHeight
+      var svg = d3.select('#graphContainer').append('svg')  
+        .attr('id', 'svgDiagram')
+        .attr('width', width)
+        .attr('height', height)
+        .call(d3.behavior.zoom().on('zoom', function () {
+        svg.attr('transform', 'translate(' + d3.event.translate + ')' + ' scale(' + d3.event.scale + ')')
+      }))
+      .append('g')
+
+      svg.append('defs').selectAll('marker')
+      .data(['suit', 'licensing', 'resolved'])
+      .enter().append('marker')
+      .attr('id', function(d) { return d })
```

Comment:
I've split it out in smaller parts - i'll consider the mixin path, but I'm not sure if it'll work properly, since i need both the Vue instance (for requestService calls) and the 'this' state of the d3 instance. Hence the let _this = this in the beginning. Will do a little test later.

Reply:
Noted